src/tools/ceph_objectstore_tool.cc

```diff
@@ -861,14 +861,24 @@ int get_attrs(
 	  cout << "snapset " << snapset << std::endl;
 	  if (!snapset.is_legacy()) {
 	    for (auto& p : snapset.clone_snaps) {
-	      hobject_t clone = hoid.hobj;
-	      clone.snap = p.first;
+	      ghobject_t clone = hoid;
+	      clone.hobj.snap = p.first;
 	      set<snapid_t> snaps(p.second.begin(), p.second.end());
+	      if (!store->exists(coll, clone)) {
```

Comment:
You could clean-up last_clones since it is not used.  last_clones is cleared when we import the head object which would be the point where it could have been useful.  It is better to check the store since you might import into a pg that already has existing objects.